development/external-plugins/automerge-notification/main.go

```diff
@@ -0,0 +1,263 @@
+// These flags are parsed along with flags defined for used clients.
+func (o *AutomergeConfig) AddFlags(fs *flag.FlagSet) {
+	fs.StringVar(&o.PubsubTopic, "pubsub-topic", "automerge", "PubSub topic to publish automerge event.")
+}
+
+// Plugin help description. This is published on Prow status plugin catalog.
+func helpProvider(_ []config.OrgRepo) (*pluginhelp.PluginHelp, error) {
+	ph := &pluginhelp.PluginHelp{
+		Description: "The automerge-notification plugin send slack notification to owners about PRs merged without required review.",
+	}
+	return ph, nil
+}
+
+// checkIfEventSupported check conditions PR must meet to send notification.
+// At the time a conditions are hard coded. In future this will be taken from Tide queries.
+func checkIfEventSupported(pr github.PullRequestEvent) bool {
+	if pr.PullRequest.Merged {
+		if pr.PullRequest.User.Login == "dependabot[bot]" || pr.PullRequest.User.Login == "kyma-bot" {
+			if github.HasLabel("skip-review", pr.PullRequest.Labels) {
+				return true
+			}
+			return true
+		}
+	}
+	return false
+}
+
+// pullRequestEventHandler process pull_request event webhooks received by plugin.
+func pullRequestEventHandler(server *externalplugin.Plugin, event externalplugin.Event) {
+	logger, atom := consolelog.NewLoggerWithLevel()
+	defer logger.Sync()
+	atom.SetLevel(pluginOptions.LogLevel)
+	logger = logger.With(externalplugin.EventTypeField, event.EventType, github.EventGUID, event.EventGUID)
+
+	var pr github.PullRequestEvent
+	if err := json.Unmarshal(event.Payload, &pr); err != nil {
+		logger.Errorw("Failed unmarshal json payload.", "error", err.Error())
+	}
+	logger = logger.With("pr-number", pr.Number,
+		"pr-sender", pr.Sender.Login)
+
+	switch pr.Action {
+	case github.PullRequestActionClosed:
+		if checkIfEventSupported(pr) {
+			// Set pubsub message payload values.
+			mergeMsgPayload := AutoMergeMessagePayload{
+				PullRequestOrg:    gogithub.String(pr.Repo.Owner.Login),
+				PullRequestRepo:   gogithub.String(pr.Repo.Name),
+				PullRequestTitle:  gogithub.String(pr.PullRequest.Title),
+				PullRequestAuthor: gogithub.String(pr.Sender.Login),
+				PullRequestNumber: gogithub.Int(pr.Number),
+				PullRequestURL:    gogithub.String(pr.PullRequest.HTMLURL),
+			}
+			pr.GUID = event.EventGUID
+			// Load repository owners.
+			owners, err := repoOwnersClient.LoadRepoOwners(pr.Repo.Owner.Login, pr.Repo.Name, "main")
+			if err != nil {
+				logger.Errorw("Failed load RepoOwners", "error", err.Error())
+			}
+			// Get git client for repository.
+			_, repoBase, err := gitClientFactory.GetGitRepoClient(pr.Repo.Owner.Login, pr.Repo.Name)
+			if err != nil {
+				logger.Errorw("Failed get repository base directory", "error", err.Error())
+			}
+			// Load repository owner aliases.
+			repoAliases, err := repoOwnersClient.LoadRepoAliases(repoBase, "OWNERS_ALIASES")
+			if err != nil {
+				logger.Errorw("failed load aliases file", "error", err.Error())
```

Comment:
Are you sure you need to call `err.Error()` directly? I assume this function accepts `interface{}`, so you don't really need to call the function directly. Are there any requirements to do it that way, instead of just using `err`?

Reply:
I'm not sure. I didn't test passing err only. I've passed string error representation because development/gcp/pkg/logging/client.go logger methods accepting kyseAndValues expect string. In case we would use this logger here, there would be no need to change calls to log message methods.

Reply:
It's just only to know how does the implementation in zap looks like. If it uses fmt package in underlying functions you don't really need to call `Error()` directly. I would suggest testing that out, since not calling the function directly looks much better.

Reply:
There is no difference if passing err or err.Error().